### ISSUE TITLE
Investigate the workflow around WorkListSource

### ISSUE THREAD

**Issue by yehee** [eddieantonio/predictive-text-studio#283]
Could we have a new interface that extends WordListSource?
```ts
interface IndexedWordListSource extends WordListSource {
  /** Autogenerated primary key */
  id: number;
}
```
This way we can know for certain if an object has the `id` property or not. IIRC TypeScript is also able to pass in `IndexedWordListSource` objects into functions that have `WordListSource` parameters without needing any explicit casting

_Originally posted by @ChrisChrisLoLo in https://github.com/eddieantonio/predictive-text-studio/pull/278#discussion_r598328858_

**Comment by eddieantonio** [comment]
> IIRC TypeScript is also able to pass in `IndexedWordListSource` objects into functions that have `WordListSource` parameters without needing any explicit casting

Yep! This works with both [nominal](https://en.wikipedia.org/wiki/Nominal_type_system) and [structural subtyping](https://en.wikipedia.org/wiki/Structural_type_system), the later of which [TypeScript implements](https://www.typescriptlang.org/docs/handbook/type-compatibility.html).
